lib/maths/analytics/CBoostedTreeFactory.cc

```diff
@@ -130,6 +130,54 @@ double minBoundary(const CBoostedTreeHyperparameters::TDoubleParameter& paramete
     T minBoundary{maxBoundary - interval};
     return parameter.fromSearchValue(minBoundary);
 }
+
+template<typename streambuf>
+class peekbuf : public streambuf {
```

Comment:
It's mega confusing to have the template argument and the base class have the same name.

Reply:
Removed - not needed